src/com/cdd/bao/util/Util.java

```diff
@@ -289,6 +289,17 @@ public static String arrayStr(int[][] arr)
 		return str;
 	}
 	
+	/**
+	 * Converts an array to a human-readable string, for debugging purposes.
+	 */
+	public static String arrayStr(Object[] arr) 
+	{
+		if (arr == null) return "{null}";
+		String str = "";
+		for (int n = 0; n < arr.length; n++) str += arr[n] == null ? "{null}" : "{" + arr[n] + "}";
+		return str;
```

Comment:
```
StringJoiner sj = new StringJoiner("}{", "{", "}");
for (Object o : arr) sj.add(o == null ? "null" : o.toString();
return sj.toString();
```

Reply:
That looks like a neat function, and I'll try to remember to use it. The `arrayStr` methods are only there for debugging, but no harm in making them better.